src/test/java/fr/insee/pogues/transforms/visualize/PoguesJSONToPoguesJSONDerefImplTest.java

```diff
@@ -0,0 +1,250 @@
+        String fooSurveyName = "FOO_SURVEY_NAME";
+        //
+        PoguesJSONToPoguesJSONDerefImpl deref = new PoguesJSONToPoguesJSONDerefImpl();
+        //
+        assertThrows(NullPointerException.class, () ->
+                deref.transform(null, fooOutputStream, fooParams, fooSurveyName));
+        assertThrows(NullPointerException.class, () ->
+                deref.transform(fooInputStream, null, fooParams, fooSurveyName));
+        assertThrows(NullPointerException.class, () ->
+                deref.transform((InputStream) null, fooParams, fooSurveyName));
+        assertThrows(NullPointerException.class, () ->
+                deref.transform((String) null, fooParams, fooSurveyName));
+        assertThrows(NullPointerException.class, () ->
+                deref.transformAsQuestionnaire(null));
+    }
+
+    @Test
+    void transform_needsDerefFalse() throws Exception {
+        //
+        String mocked = "some json questionnaire";
+        Map<String, Object> fooParams = Map.of("needDeref", false);
+        String fooSurveyName = "FOO_SURVEY_NAME";
+        //
+        PoguesJSONToPoguesJSONDeref deref = new PoguesJSONToPoguesJSONDerefImpl();
+        //
+        assertEquals(mocked, deref.transform(mocked, fooParams, fooSurveyName));
+    }
+
+    /**
+     * The questionnaire 'lct78jr8' contains references to other questionnaires: 'l4i3m6qa', 'l6dnlrka' and 'lct8pcsy'.
+     * These will be mocked as null, but the method should still return a questionnaire.
+     * */
+    @Test
+    void dereference_nullReferences() throws Exception {
+        // Given
+        // Read tested questionnaire
+        URL url = this.getClass().getClassLoader().getResource(
+                "transforms/PoguesJSONToPoguesJSONDeref/filter_and_loop/lct78jr8.json");
+        assert url != null;
+        String testedInput = Files.readString(Path.of(url.toURI()));
+        // Mock questionnaire service
+        QuestionnairesService questionnairesService = Mockito.mock(QuestionnairesService.class);
+        Mockito.when(questionnairesService.getQuestionnaireByID("l4i3m6qa")).thenReturn(null);
+        Mockito.when(questionnairesService.getQuestionnaireByID("l6dnlrka")).thenReturn(null);
+        Mockito.when(questionnairesService.getQuestionnaireByID("lct8pcsy")).thenReturn(null);
+
+        // When
+        PoguesJSONToPoguesJSONDerefImpl deref = new PoguesJSONToPoguesJSONDerefImpl(questionnairesService);
+        Questionnaire outQuestionnaire = deref.transformAsQuestionnaire(testedInput);
+
+        // Then
+        assertNotNull(outQuestionnaire);
+    }
+
+    /**
+     * Tested questionnaire: 'lct78jr8'
+     * This questionnaire contains references to the following questionnaires:
+     * - 'l4i3m6qa' (with a filter condition in the root questionnaire)
+     * - 'l6dnlrka' (simple questionnaire)
+     * - 'lct8pcsy' (including loops) */
+    @Test
+    void dereference_filterAndLoop() throws Exception {
```

Comment:
seems to have multiple different tests here. Could be divide into several methods for each test type

Reply:
Tests in this class are more like integration tests for the de-referencing feature (given some questionnaire files, test as much things as possible with their content). Unit tests are on the `QuestionnaireComposition` class that holds the logic.